openerp/addons/base/ir/ir_translation.py

```diff
@@ -283,21 +283,31 @@ def _get_ids(self, cr, uid, name, tt, lang, ids):
     def _set_ids(self, cr, uid, name, tt, lang, ids, value, src=None):
         self._get_ids.clear_cache(self)
         self._get_source.clear_cache(self)
-
-        cr.execute('delete from ir_translation '
+        original_module = self.pool[name.split(',')[0]]._original_module
+        cr.execute('update ir_translation '
+                  'set value=%s '
+                  '  , src=%s '
+                  '  , state=%s '
                 'where lang=%s '
                     'and type=%s '
                     'and name=%s '
-                    'and res_id IN %s',
-                (lang,tt,name,tuple(ids),))
-        for id in ids:
+                    'and module=%s '
```

Comment:
Filtering on `module` is wrong and breaks translations because it creates an asymmetry between `_get_ids` and `_set_ids`. `_get_ids` loads any translation regardless of the `module` field, but `_set_ids` will fail to see/update them. So you easily get in a situation where editing a translatable field seems impossible because it updates one translation (the one with `module` set) but reads the other first.
This is even more likely because opening the translation interface in the UI calls `ir_translation.translate_fields()`, which bootstraps translations without setting the `module` (because there is none to be set).

Regarding the bug in the PR description: it seems the right place to fix it should be the import of translations when a module is installed/updated, to make sure it never leads to duplicate entries (by respecting the same criterions as `get_ids` too). Perhaps it would be as simple as deleting [this line](https://github.com/odoo/odoo/blob/39d17c2/openerp/addons/base/ir/ir_translation.py#L126).
 @rven @mart-e could you verify whether that would fix your bug?

In the mean time we'll have to partially revert the code related to the `module` in `set_ids`. It should not be necessary anyway: if the `module` is set, we preserve it, otherwise it's a new manual translation that should not belong to any module.